### PR TITLE
S12_tabs

### PR DESCRIPTION
**Pull request by valntyn**
[DEMO](https://valntyn.github.io/products-e-commerce/#/)

### CODE REVIEW

File: src/components/ProductBlock/ProductDesc/ProductTabs/Tabs/index.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      if (selectedTab) {
+        const { width, left } = selectedTab.getBoundingClientRect();
+        const containerLeft = tabRef.current?.getBoundingClientRect().left || 0;
+
+        setAnimateStyle({
+          width,
+          transform: `translateX(${left - containerLeft}px)`,
+        });
+      }
+    };
```
Comment (Yuliia143) [inline src/components/ProductBlock/ProductDesc/ProductTabs/Tabs/index.tsx]: Don't describe function in useEffect

File: src/components/ProductBlock/ProductDesc/ProductTabs/Tabs/index.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    recalculatePosition();
+
+    const handleResize = () => {
+      recalculatePosition();
+    };
```
Comment (Yuliia143) [inline src/components/ProductBlock/ProductDesc/ProductTabs/Tabs/index.tsx]: ?

File: src/components/ProductBlock/ProductDesc/ProductTabs/Tabs/index.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, [selectedId]);
+
+  return (
+    <div className={classNames('tabs', className)} ref={tabRef}>
+      {tabs
```
Comment (Yuliia143) [inline src/components/ProductBlock/ProductDesc/ProductTabs/Tabs/index.tsx]: tabs?.map

File: src/components/ProductBlock/ProductDesc/ProductTabs/Tabs/index.tsx (hunk 2 above)
Comment (valntyn) [reply src/components/ProductBlock/ProductDesc/ProductTabs/Tabs/index.tsx]: i've wrote in this way that the strip under tabs worked correctly
but I'll rewrite it

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**